File: backend/src/main/java/webshop/backend/domains/product/Product.java
package webshop.backend.domains.product;

import jakarta.persistence.*;
import lombok.*;
import webshop.backend.domains.category.Category;

import java.math.BigDecimal;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "products")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String description;
    private BigDecimal price;
    private int stock;

    @ManyToOne
    private Category category;
}

File: vue-frontend/app/types/product.ts
export interface Product {
  id: number;
  name: string;
  description: string;
  price: number;
  stock: number;
  categoryId: number;
  categoryName: string;
}

export interface ProductResponseDto {
  id: number;
  name: string;
  description: string;
  price: number;
  stock: number;
  categoryId: number;
  categoryName: string;
}

export interface ProductRequestDto {
  name: string;
  description: string;
  price: number;
  stock: number;
  categoryId: number;
}